test/common/upstream/upstream_impl_test.cc

```diff
@@ -973,7 +985,8 @@ TEST_F(HostSetImplLocalityTest, MissingWeight) {
   HostsPerLocalitySharedPtr hosts_per_locality =
       makeHostsPerLocality({{hosts_[0]}, {hosts_[1]}, {hosts_[2]}});
   LocalityWeightsConstSharedPtr locality_weights{new LocalityWeights{1, 0, 1}};
-  host_set_.updateHosts({}, {}, hosts_per_locality, hosts_per_locality, locality_weights, {}, {});
+  host_set_.updateHosts({}, std::make_shared<const HostVector>(hosts_per_locality->get()[0]),
```

Comment:
These tests are a bit weird, in that you have healthy hosts that aren't a subset of the host list (first arg).